parse_node/parse_source_file.ts

```diff
@@ -21,18 +28,78 @@ const getClassDeclarationHeader = (
 
     const clause = node.heritageClauses[0] as ts.HeritageClause
     const type = clause.types[0]
+    const tsType = props.program.getTypeChecker().getTypeAtLocation(type)
 
     extendsFrom = type.getText()
+
+    if (tsType.symbol && tsType.symbol.declarations) {
+      // Handle the case when extended class is a inner clas of file
+      const classDecl = tsType.symbol.declarations.find((v) =>
+        ts.isClassDeclaration(v)
+      ) as ClassDeclaration | null
+
+      if (classDecl) {
+        const modifiers = (classDecl.modifiers ?? []).map((v) => v.getText())
+
+        const asset = props.project
+          .sourceFiles()
+          .find((v) => v.fsPath === classDecl.getSourceFile().fileName)
+
+        if (
+          !asset &&
+          !modifiers.includes("declare") &&
+          (!modifiers.includes("default") || !classDecl.name)
+        ) {
+          addError({
+            description:
+              "Class extends a type for which a source file is missing. This is an internal ts2gd bug. Please report it.",
+            error: ErrorName.ClassDoesntExtendAnything,
+            location: node,
+            stack: new Error().stack ?? "",
+          })
+        } else if (
+          !modifiers.includes("declare") &&
+          !modifiers.includes("default")
+        ) {
+          // If a class declaration does not have default export then this is an inner class
+          // The syntax for extending inner class in gdscript is: extends "res://compiled/Test.gd".BaseType
+
+          extendsFrom = asset
+            ? props.sourceFileAsset === asset
+              ? type.getText()
+              : `"${asset.resPath}".${type.getText()}`
+            : "[missing]"
```

Comment:
We should show an error in ts2gd rather than emitting invalid code here.

Reply:
I have refactor the code so it adds an error instead.